EMEA-Day3-ExternalIdentities-AppIntegrationMSAL.md

```diff
@@ -0,0 +1,136 @@
+
+A Single-Page Application (SPA) calling a Web API.
+The Web API is the MSAL Node JS Web API.
+We will run both at the same time, in order to see the middleware library (Passport.js) validating the token. This will show the content learned early in the OAuth2.0 & OIDC sections.
+
+### Run the web API
+
+- `git clone https://github.com/Azure-Samples/active-directory-b2c-javascript-nodejs-webapi.git`
+
+- `cd active-directory-b2c-javascript-nodejs-webapi`
+
+- Install Node.Js if you haven't already
+
+- Install and update the Node dependencies
+
+    `npm install && npm update`
+
+- Run the Web API. By default, it runs on `http:localhost:5000`
+
+    `npm start`
+
+### Run the web app
+
+- `git clone https://github.com/Azure-Samples/active-directory-b2c-javascript-angular-spa.git`
+
+- `cd active-directory-b2c-javascript-angular-spa`
+
+- Install and update the Node dependencies
+
+    `npm install && npm update`
+
+- Open `src/app/app-config.ts`
+- Edit Line 49:
+
+    **replace**: `webApi: "https://fabrikamb2chello.azurewebsites.net/hello"`
+
+    **with**: `webApi: "http://localhost:5000/hello"`
+- Hit Save
+
+- Run the Web app.
+
+    `npm start`
+
+    `Listening on port 6420...`
+
+Visit `http://localhost:6420` and perform the following actions:
+
+1) Click the Login button to start the Azure AD B2C sign in or sign up workflow.
+1) Once signed in, you can click the Call Web API button to have your display name returned from the Web API call as a JSON object.
+1) Click "Logout" to logout from the application.
+
+
+## Lab 02: Create your own web app
+
+### Install the Microsoft.Identity.Web templates in .NET Core
+
+In this lab, we'll need to use the Microsoft.Identity.Web project templates (which are also available in .NET 5.0 from Preview 7). To install them in your system:
+
+1. Download the [Microsoft.Identity.Web.ProjectTemplates-1.1.0](https://www.nuget.org/api/v2/package/Microsoft.Identity.Web.ProjectTemplates/1.1.0) NuGet package from NuGet.org, and save it locally
+1. In a developer command prompt, type
+   ```dotnetcli
+   dotnet new -i Microsoft.Identity.Web.ProjectTemplates.1.1.0.nupkg 
+   ```
+
+This installs the templates, and displays which project templates are available. (You can also type `dotnet new --help` to see the list)
+
+### Create a B2C web app that sign-in users
+
+In a developer command prompt:
+
+1. Create a folder, and cd into it.
+   ```bash
+   mkdir webapp
+   cd webapp
+   ```
+
+1. Create the B2C web app
+   ```dotnetcli
+   dotnet new webapp2 --auth IndividualB2C ^
+                     --aad-b2c-instance "https://fabrikamb2c.b2clogin.com" ^
+                     --domain "fabrikamb2c.onmicrosoft.com" ^
+                     --client-id "fdb91ff5-5ce6-41f3-bdbd-8267c817015d"
+   ```
+
+1. Edit the `webapp.csproj` file to replace `<TargetFrameworks>netcoreapp3.1; net5.0</TargetFrameworks>` by `<TargetFramework>netcoreapp3.1</TargetFramework>` (unless you have installed the .NET 5.0 preview framework)
```

Comment:
This step is no longer needed: the project templates are now netcoreapp3.1 only (as net5.0 ships with our templates)